src/msg/async/rdma/Infiniband.cc

```diff
@@ -30,7 +30,8 @@ static const uint32_t MAX_INLINE_DATA = 0;
 static const uint32_t TCP_MSG_LEN = sizeof("0000:00000000:00000000:00000000:00000000000000000000000000000000");
 static const uint32_t CQ_DEPTH = 30000;
 
-Port::Port(CephContext *cct, struct ibv_context* ictxt, uint8_t ipn): ctxt(ictxt), port_num(ipn), gid_idx(0)
+Port::Port(CephContext *cct, struct ibv_context* ictxt, uint8_t ipn): ctxt(ictxt), port_num(ipn),
+  gid_idx(cct->_conf->ms_async_rdma_gid_idx)
```

Comment:
if the constructor of `Port` is not in the critical path, maybe we can use `m_cct->_conf.get_val<int64_t>` instead? BTW, should this variable be unsigned ?

Reply:
gid_idx is defined as "int gid_idx" and it's used as the argument to call below API
int ibv_query_gid(struct ibv_context *context, uint8_t port_num, **int index**, union ibv_gid *gid)
It should not be unsigned.